presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchClient.java

```diff
@@ -101,9 +111,39 @@ public ElasticsearchClient(ElasticsearchTableDescriptionProvider descriptions, E
 
         for (ElasticsearchTableDescription tableDescription : tableDescriptions.getAllTableDescriptions()) {
             if (!clients.containsKey(tableDescription.getClusterName())) {
-                Settings settings = Settings.builder().put("cluster.name", tableDescription.getClusterName()).build();
+                Settings settings;
                 TransportAddress address = new TransportAddress(InetAddress.getByName(tableDescription.getHost()), tableDescription.getPort());
-                TransportClient client = new PreBuiltTransportClient(settings).addTransportAddress(address);
+                TransportClient client;
```

Comment:
We can simplify the constructor by extracting a factory method here:

```
    private static TransportClient createTransportClient(ElasticsearchConnectorConfig config, TransportAddress address)
    {
        Settings settings;
        TransportClient client;
        switch (config.getCertificateFormat()) {
            // ....
        }
        return client;
    }
```

Then in the constructor: `TransportClient client = createTransportClient(config, address);`